src/cascadia/TerminalApp/TerminalWindow.cpp

```diff
@@ -0,0 +1,1262 @@
+    USES_RESOURCE(L"UnknownTheme"),
+    USES_RESOURCE(L"DuplicateRemainingProfilesEntry"),
+};
+
+static_assert(settingsLoadWarningsLabels.size() == static_cast<size_t>(SettingsLoadWarnings::WARNINGS_SIZE));
+// Errors are defined in AppLogic.cpp
+
+// Function Description:
+// - General-purpose helper for looking up a localized string for a
+//   warning/error. First will look for the given key in the provided map of
+//   keys->strings, where the values in the map are ResourceKeys. If it finds
+//   one, it will lookup the localized string from that ResourceKey.
+// - If it does not find a key, it'll return an empty string
+// Arguments:
+// - key: the value to use to look for a resource key in the given map
+// - map: A map of keys->Resource keys.
+// Return Value:
+// - the localized string for the given type, if it exists.
+template<typename T>
+winrt::hstring _GetMessageText(uint32_t index, const T& keys)
+{
+    if (index < keys.size())
+    {
+        return GetLibraryResourceString(til::at(keys, index));
+    }
+    return {};
+}
+
+// Function Description:
+// - Gets the text from our ResourceDictionary for the given
+//   SettingsLoadWarning. If there is no such text, we'll return nullptr.
+// - The warning should have an entry in settingsLoadWarningsLabels.
+// Arguments:
+// - warning: the SettingsLoadWarnings value to get the localized text for.
+// Return Value:
+// - localized text for the given warning
+static winrt::hstring _GetWarningText(SettingsLoadWarnings warning)
+{
+    return _GetMessageText(static_cast<uint32_t>(warning), settingsLoadWarningsLabels);
+}
+
+// Function Description:
+// - Creates a Run of text to display an error message. The text is yellow or
+//   red for dark/light theme, respectively.
+// Arguments:
+// - text: The text of the error message.
+// - resources: The application's resource loader.
+// Return Value:
+// - The fully styled text run.
+static Documents::Run _BuildErrorRun(const winrt::hstring& text, const ResourceDictionary& resources)
+{
+    Documents::Run textRun;
+    textRun.Text(text);
+
+    // Color the text red (light theme) or yellow (dark theme) based on the system theme
+    auto key = winrt::box_value(L"ErrorTextBrush");
+    if (resources.HasKey(key))
+    {
+        auto g = resources.Lookup(key);
+        auto brush = g.try_as<winrt::Windows::UI::Xaml::Media::Brush>();
+        textRun.Foreground(brush);
+    }
+
+    return textRun;
+}
+
+namespace winrt::TerminalApp::implementation
+{
+    TerminalWindow::TerminalWindow(const TerminalApp::SettingsLoadEventArgs& settingsLoadedResult) :
+        _settings{ settingsLoadedResult.NewSettings() },
+        _initialLoadResult{ settingsLoadedResult }
+    {
+        // The TerminalPage has to be constructed during our construction, to
+        // make sure that there's a terminal page for callers of
+        // SetTitleBarContent
+        _root = winrt::make_self<TerminalPage>();
+        _root->WindowProperties(*this);

```

Comment:
q from before answered: I now understand why it is an Interface

Reply:
Note that this creates a circular reference between `TerminalWindow` and `TerminalPage`

Reply:
Resolved in 44b238e which greatly improved everything with that